src/modules/promote.ts

```diff
@@ -26,11 +27,17 @@ const throwIfIsMaster = (workspace: string) => {
 }
 
 const handleConflict = async () => {
-  const conflictsFound = await checkForConflicts()
+  const conflictsFound = await vbase.checkForConflicts()
 
   if (conflictsFound) {
     await axios.get(workspaceUrl)
   }
+
+  // @vtex/api expects a full implementation of the client, so we need to cast it to any.
+  // The partial implementation is enough to solve conflicts.
+  const conflictsResolver = new MineWinsConflictsResolver((vbase as Partial<VBase>) as any, 'userData', '')
```

Comment:
Quick questions to get my mind around here:

1. The purpose of `axios.get(workspaceUrl)` is to perform a rebase, is that right? Should this logic also be inside the `conflictsFound` branch (maybe replace the get)? The resolver conflict operation is a expensive one for db so it would be good if we can limit its usage.
2. This should add some latency to the operation. Did you feel the promote command any slower when testing it?
3. The default strategy to resolver the entries on db is `masterWins`. Isn't it conflicting using `mineWins` here?

Reply:
1. Yes. I was thinking of removing this .get, but I did not know the side effects of doing this since I'm just aware it's forcing the rebase. It might be there for other purposes. I will add the new logic inside the checkForConflict condition!
2. I didn't notice any latency; it's looking very similar.
3. I believe the "optimal" solution would be defining it by module, but since it is a general strategy, we need this workaround here(that already happened today). Besides that, the mineWins runs inside the JSON, not in the DB, so they are made at different levels(file x actual data).

Reply:
About #3, I think would be nice to add some comments in the code explaining the logic behind it. Someone in the future looking at it might be confused about this mix of strategies. If I got it right, when someone accesses a workspace, this rebase in the JSON level happens with the mineWins strategy. Is that true?

Reply:
Agree, I will add it. That's correct!